polaris-react/src/utilities/use-index-resource-state.ts

```diff
@@ -41,12 +42,37 @@ export function useIndexResourceState<T extends {[key: string]: unknown}>(
     resourceFilter: undefined,
   },
 ) {
-  const [selectedResources, setSelectedResources] = useState(
-    initSelectedResources,
-  );
+  const [selectedResources, setSelectedResources] = useState<
+    ReadonlySet<string>
+  >(new Set(preCheckedResources));
   const [allResourcesSelected, setAllResourcesSelected] = useState(
     initAllResourcesSelected,
   );
+  const [unselectedResources, setUnselectedResources] = useState<
```

Comment:
This state is needed for `Test drive`.

We need to know all the "stores" that were unselected. In our UX, we pre-check every store and if they become unselected we must update them on the backend.